### ISSUE TITLE
JaCoCo Build fails with Maven 3.8.2

### ISSUE THREAD

**Issue by marchof** [jacoco/jacoco#1218]
Since Maven 3.8.2 the JaCoCo build fails with the following error:

    org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-plugin-plugin:3.6.0:report (report) on project jacoco-maven-plugin: Execution report of goal org.apache.maven.plugins:maven-plugin-plugin:3.6.0:report failed: A required class was missing while executing org.apache.maven.plugins:maven-plugin-plugin:3.6.0:report: org/jacoco/report/check/IViolationsOutput

Therefore the Azure CI breaks, see for example PR #1217.

**Comment by marchof** [comment]
Relevant stack trace probably is:

```
Caused by: java.lang.ClassNotFoundException: org.jacoco.report.check.IViolationsOutput
    at java.net.URLClassLoader.findClass (URLClassLoader.java:471)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:589)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:522)
    at java.lang.ClassLoader.defineClass1 (Native Method)
    at java.lang.ClassLoader.defineClass (ClassLoader.java:1017)
    at java.security.SecureClassLoader.defineClass (SecureClassLoader.java:174)
    at java.net.URLClassLoader.defineClass (URLClassLoader.java:550)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:458)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:452)
    at java.security.AccessController.doPrivileged (Native Method)
    at java.net.URLClassLoader.findClass (URLClassLoader.java:451)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:589)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:522)
    at java.lang.Class.forName0 (Native Method)
    at java.lang.Class.forName (Class.java:398)
    at org.apache.maven.tools.plugin.generator.GeneratorUtils.isMavenReport (GeneratorUtils.java:699)
    at org.apache.maven.tools.plugin.generator.PluginXdocGenerator.writeReportNotice (PluginXdocGenerator.java:240)
    at org.apache.maven.tools.plugin.generator.PluginXdocGenerator.writeBody (PluginXdocGenerator.java:188)
    at org.apache.maven.tools.plugin.generator.PluginXdocGenerator.processMojoDescriptor (PluginXdocGenerator.java:138)
    at org.apache.maven.tools.plugin.generator.PluginXdocGenerator.execute (PluginXdocGenerator.java:113)
    at org.apache.maven.plugin.plugin.PluginReport.generateMojosDocumentation (PluginReport.java:423)
    at org.apache.maven.plugin.plugin.PluginReport.executeReport (PluginReport.java:272)
    at org.apache.maven.reporting.AbstractMavenReport.generate (AbstractMavenReport.java:255)
    at org.apache.maven.reporting.AbstractMavenReport.execute (AbstractMavenReport.java:143)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

**Comment by Godin** [comment]
For me seems that this might be related to the following issue described in [Maven 3.8.2 release notes](https://maven.apache.org/docs/3.8.2/release-notes.html)

> ### Known Issues
> * If any of your plugin mojos spawns new threads there might be a change in class loading as a result of the fix for [MNG-6843](https://issues.apache.org/jira/browse/MNG-6843). See [discussion](https://lists.apache.org/thread.html/r0777c9e364f93a609cb4c3da6e634139b9c400166e280856ee25ba72%40%3Cdev.maven.apache.org%3E) with a possible fix for your plugin mojo as well as a potential general fix in [MNG-7212](https://issues.apache.org/jira/browse/MNG-7212).

@rfscholte @olamy could you please help - am I right? and so we'll have to wait fix of MNG-7212 before will be able to use Maven versions newer than 3.8.1?

@marchof meanwhile I propose to use Maven 3.8.1 for CI builds - https://github.com/jacoco/jacoco/pull/1221

**Comment by rfscholte** [comment]
I would have the same guess

**Comment by Godin** [comment]
@rfscholte Thanks!

And while [MNG-7212](https://issues.apache.org/jira/browse/MNG-7212) wasn't yet resolved, seems that Maven 3.8.3 solves the issue that we encountered - #1229 👍 ❤️ which I guess is related to resolution of [MNG-7251](https://issues.apache.org/jira/browse/MNG-7251)
